app/src/main/java/ingsw/group1/findmyphone/log/LogItemSwipeCallback.java

```diff
@@ -0,0 +1,163 @@
+
+import ingsw.group1.findmyphone.R;
+import ingsw.group1.findmyphone.activity.NavHolderActivity;
+import ingsw.group1.findmyphone.log.holders.LogViewHolder;
+import ingsw.group1.findmyphone.log.items.LogItem;
+import it.xabaras.android.recyclerview.swipedecorator.RecyclerViewSwipeDecorator;
+
+/**
+ * Class defining a Delete swipe gesture for a LogRecyclerAdapter.
+ *
+ * @author Riccardo De Zen.
+ */
+public class LogItemSwipeCallback extends ItemTouchHelper.SimpleCallback {
+
+    @NonNull
+    private final LogRecyclerAdapter logAdapter;
+    @NonNull
+    private final NavController navController;
+
+    private final String DELETION_MESSAGE;
+    private final String UNDO_BUTTON;
+    private final String DELETE_LABEL;
+    private final String ADD_LABEL;
+
+    private final int LEFT_COLOR;
+    private final int RIGHT_COLOR;
+
+    public LogItemSwipeCallback(@NonNull LogRecyclerAdapter adapter,
+                                @NonNull Resources resources,
+                                @NonNull NavController navController) {
+        super(0, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT);
+        this.logAdapter = adapter;
+        this.navController = navController;
+        DELETION_MESSAGE = resources.getString(R.string.delete_item_message);
+        UNDO_BUTTON = resources.getString(R.string.undo_button);
+        DELETE_LABEL = resources.getString(R.string.delete_label);
+        ADD_LABEL = resources.getString(R.string.add_label);
+        LEFT_COLOR = resources.getColor(R.color.deleteColor);
+        RIGHT_COLOR = resources.getColor(R.color.modifyColor);
+    }
+
+    @Override
+    public int getSwipeDirs(@NonNull RecyclerView recyclerView,
+                            @NonNull RecyclerView.ViewHolder viewHolder) {
+        // If the holder does not require the right action then only left is returned.
+        if ((viewHolder.getItemViewType() & LogItem.NAMELESS) != LogItem.NAMELESS)
+            return ItemTouchHelper.LEFT;
+        return super.getSwipeDirs(recyclerView, viewHolder);
+    }
+
+    /**
+     * Move events are not handled.
+     *
+     * @param recyclerView The Recycler view triggering the event.
+     * @param viewHolder   The moved ViewHolder.
+     * @param target       The ViewHolder that gets replaced.
+     * @return {@code false} meaning the event was not handled.
+     */
+    @Override
+    public boolean onMove(@NonNull RecyclerView recyclerView,
+                          @NonNull RecyclerView.ViewHolder viewHolder,
+                          @NonNull RecyclerView.ViewHolder target) {
+        return false;
+    }
+
+    /**
+     * When the item is swiped left, it is removed from the adapter.
+     *
+     * @param viewHolder The ViewHolder that triggered the swipe.
+     * @param direction  The direction in which the swipe happened.
+     */
+    @Override
+    public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
+        switch (direction) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5114829)